<?xml version="1.0"?>
<launch>
  <!-- Launch file for plane detection with Isaac Sim -->
  <!-- Use this when running with Isaac Sim simulation -->

  <!-- Set use_sim_time to true for Isaac Sim -->
  <param name="use_sim_time" value="true" />

  <!-- Plane Detection Parameters -->
  <arg name="min_plane_points" default="500" />
  <arg name="ransac_threshold" default="0.02" />
  <arg name="max_planes" default="5" />
  <arg name="min_plane_area" default="0.1" />
  <arg name="downsample_factor" default="4" />

  <!-- Plane Detection Node -->
  <node name="plane_detection_node" pkg="csc752-final-project" type="plane_detection_node.py" output="screen">
    <param name="min_plane_points" value="$(arg min_plane_points)" />
    <param name="ransac_threshold" value="$(arg ransac_threshold)" />
    <param name="max_planes" value="$(arg max_planes)" />
    <param name="min_plane_area" value="$(arg min_plane_area)" />
    <param name="downsample_factor" value="$(arg downsample_factor)" />
  </node>

</launch>
